letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -272,7 +279,8 @@ def _choose_vhost_from_list(self, target_name):
         elif not vhost.ssl:
             addrs = self._get_proposed_addrs(vhost, "443")
             # TODO: Conflicts is too conservative
-            if not any(vhost.enabled and vhost.conflicts(addrs) for vhost in self.vhosts):
+            if not any(vhost.enabled and vhost.conflicts(addrs) for vhost in self.vhosts)\
+                    and not temp:
```

Comment:
To elaborate on this, the problem is in `_choose_vhost_from_list`. If `temp` is `True`, the function doesn't return early like it does in `choose_vhost`. As a result, the line `self.assoc[target_name] = vhost` is executed. This means that a potentially non-ssl vhost is cached, which will be a problem when a ssl vhost is required in the future.

The way I'd fix this is revert the changes to the `if not any (vhost.enabled ...` line and before `elif not vhost.ssl:` line add:
```
elif temp:
    return vhost
```

Hopefully this helps! If not, feel free to ping me here or on IRC.